src/Evolution/Systems/GeneralizedHarmonic/Constraints.cpp

```diff
@@ -449,9 +450,12 @@ void four_index_constraint(
   static_assert(
       SpatialDim == 3,
       "four_index_constraint() currently only supports 3 spatial dimensions");
-  for (auto& component : *constraint) {
-    component = 0.0;
+  if (get_size(get<0, 0, 0>(*constraint)) != get_size(get<0, 0, 0, 0>(d_phi))) {
+    tnsr::iaa<DataType, SpatialDim, Frame> constraint_(get<0, 0, 0, 0>(d_phi));
```

Comment:
Same comment as I just made for the energy; I'd prefer to not have the intermediate `constraint_` variable, so you can just assign directly from the constructor:
`*constraint = tnsr::iaa<DataType, SpatialDim, Frame>{get<0, 0, 0, 0>(d_phi)};`